tests/testsuite/update.rs

```diff
@@ -2318,10 +2314,12 @@ fn update_precise_breaking_pinned() {
         .masquerade_as_nightly_cargo(&[])
         .with_status(101)
         .with_stderr_data(str![[r#"
-[ERROR] unknown `-Z` flag specified: update-precise-breaking
-
-For available unstable features, see https://doc.rust-lang.org/nightly/cargo/reference/unstable.html
-If you intended to use an unstable rustc feature, try setting `RUSTFLAGS="-Zupdate-precise-breaking"`
+[UPDATING] `dummy-registry` index
+[ERROR] failed to select a version for the requirement `pinned = "=0.1"`
+candidate versions found which didn't match: 0.2.0
```

Comment:
So `--precise` will error if the mentioned item can't upgrade but `--breaking` won't...

Reply:
Unlike with `--breaking`, there is no command line difference between a breaking and a non-breaking precise update. So we just try to do any upgrades, and whether or not we did any, the same lockfile update is run. So a precise update may fall back to being non-breaking, and may fail in the same way as a precise update could break before.

If we changed the API to `--breaking --precise` we could stop if there aren't any upgrades.

Reply:
I've renamed the existing task to

>  Should cargo update --breaking error if the command completed without doing any upgrades? 

and added this as a parallel task

> Should cargo update --precise <breaking> error if the command completed without doing any upgrades?